great_expectations/data_context/data_context_variables.py

```diff
@@ -3,17 +3,23 @@
 from dataclasses import dataclass
 from typing import Any, Optional
 
-from great_expectations.core.data_context_key import DataContextVariableKey
+from great_expectations.core.data_context_key import DataContextKey
 from great_expectations.data_context.types.base import (
     AnonymizedUsageStatisticsConfig,
     ConcurrencyConfig,
+    DataContextConfig,
     NotebookConfig,
     ProgressBarsConfig,
 )
-from great_expectations.data_context.types.resource_identifiers import GeCloudIdentifier
+from great_expectations.data_context.types.resource_identifiers import (
+    ConfigurationIdentifier,
+    GeCloudIdentifier,
+)
+from great_expectations.data_context.util import substitute_config_variable
 
 
 class DataContextVariableSchema(str, enum.Enum):
+    DATA_CONTEXT = "data_context"  # If retrieving/setting the entire config at once
```

Comment:
Our enum now includes `DATA_CONTEXT` to refer to the whole config - we need the ability to modify both individual components of the config as well as the entire config itself. 

Logic exists downstream in the stores to work with these changes.